### PR TITLE
Better motion page UX

### LINKED ISSUE #3409: Improve not enough activated tokens message + share banner
### Story

When a user arrives at a motion and wants to stake or object to the motion, if they don’t have enough or any activated tokens they are presented with the screen below. Which tells them to activate more tokens, however, there is nothing in the UI to tell them how to do so.

b61bd98e-f3fd-4739-a567-a805db979eef

### Description

To provide a better user experience when not having enough activated tokens to stake on a motion, we need to make it more intuitive for any level of user to access. To do this, we are adding an improved description, and adding an "Activate" button that will trigger the `TokenActivationPopover`.

In this feature, please also update the design of the banner as of the Figma design.

### Design

[Figma Link](https://www.figma.com/file/CCgGsBzhX3BUENS1IvgP7N/Motions-%26-Disputes?node-id=6457%3A184140)

The message in the red callout box should read "You need at least 0.10101 CLNY activated to stake on or object to this motion. You have 0 CLNY activated. Click `Activate` below to activate more.”

Not yet staked

### PR DESCRIPTION
**Pull request by chinins**
## Description

This PR:

- updates the motion banner when less than 10% was staked
- changes the U & UXI for when not enough tokens are activated

**2 changes vs the specs (both aligned with Arren)**:

- the share URL button has more margin-right - to be aligned with the avatar
- the `Activate more tokens to be eligible to stake.` is now a button (vs. just a div in the specs)

**New stuff** ✨

- add `TokenActivation` context & provider so that we can access it throughout the dapp
- use the new context in the Staking widget

**Changes** 🏗

- update staking widget, slider & errors components
- update motion banner share url button UI
- update TokenActivationPopover component

Screenshot 2022-05-31 at 17 33 46

resolves #3409

### CODE REVIEW

**Review by arrenv — CHANGES_REQUESTED** [review]
This is a great UX improvement, nice one!

A couple of points:

- The styling of the Share URL button is not showing correctly for me.
- The copy in the red callout should be "Activate {{HOW_MANY_MORE_TOKENS_REQUIRED}} more CLNY to be eligible to stake.”

827f41eb-5a8d-4f69-afe1-5afb0cabbeec

File: src/modules/dashboard/components/ActionsPage/StakeRequiredBanner/StakeRequiredBanner.css [hunk 1]
```diff
@@ -11,7 +11,41 @@
...
-  font-size: var(--size-small);
+  padding-right: 6px;
+  padding-left: 10px;
+  border: 0px;
+  color: var(--action-secondary);
+  opacity: 1;
+
+  &:hover {
+    border: 1px solid var(--action-secondary);
+    background-color: rgba(255, 255, 255);
```
Comment (danbr) [inline src/modules/dashboard/components/ActionsPage/StakeRequiredBanner/StakeRequiredBanner.css:29]: Not limited to your PR but I'm wondering why there is no var for rgba(255, 255, 255) i.e. white???

File: src/modules/dashboard/components/ActionsPage/StakingValidationError/StakingValidationError.tsx [hunk 2]
```diff
@@ -40,11 +42,31 @@ const stakeValidationMSG = defineMessage({
...
-);
+const StakingValidationError = ({ stakeType, errorValues }: Props) => {
+  const { setIsOpen: openTokenActivationPopover } = useContext(
+    TokenActivationContext,
+  );
+
+  if (stakeType === 'tokens') {
+    return (
+      <Button
+        text={stakeValidationMSG[stakeType]}
```
Comment (danbr) [inline src/modules/dashboard/components/ActionsPage/StakingValidationError/StakingValidationError.tsx:53]: As the stakeType will always be 'tokens', theres no need to perform a lookup for the MSG.

File: src/modules/users/components/TokenActivationProvider/TokenActivationProvider.tsx [hunk 3]
```diff
@@ -0,0 +1,30 @@
+import React, { createContext, useState, ReactNode } from 'react';
+
+export const TokenActivationContext = createContext<{
+  isOpen?: boolean;
```
Comment (danbr) [inline src/modules/users/components/TokenActivationProvider/TokenActivationProvider.tsx:4]: Not sure this should be optional?

**Review by danbr — CHANGES_REQUESTED** [review]
Looks good. - the hard part is working well. I made a few comments, and a couple of issues to address.

Screenshot 2022-06-02 at 07 30 11


@arrenv More generally, I find the UI design bizarre!! I dont understand the need for 2 activation buttons (red & green) when they both perform the same action of opening the dialog? Have i misunderstood?

Perhaps a tooltip on the activation button could be helpful to explain.

**Comment by arrenv** [comment]
> @arrenv More generally, I find the UI design bizarre!! I dont understand the need for 2 activation buttons (red & green) when they both perform the same action of opening the dialog? Have i misunderstood?

@danbr The spec didn't require the red callout to be a button. This was a suggestion that does not really cause any harm and can enhance the user experience in a potentially confusing moment. However, the green button was part of the spec as it is consistent within the UX, if it was only the red banner it would not be immediately clear for a beginner that it is clickable.

File: src/modules/dashboard/components/ActionsPage/StakingValidationError/StakingValidationError.tsx (hunk 2 above)
Comment (chinins) [reply src/modules/dashboard/components/ActionsPage/StakingValidationError/StakingValidationError.tsx:53]: fixed

**Comment by chinins** [comment]
I've fixed everything mentioned in the comments except the `Share URL` button text & icon alignment. I can't replicate so can't fix.
@arrenv, @ArmandoGraterol, could you please play with it and suggest the fix that works for you? I will check that it works on my machine add it to the PR. Thank you in advance - I would really appreciate that :)

File: src/modules/dashboard/components/ActionsPage/StakingValidationError/StakingValidationError.tsx [hunk 4]
```diff
@@ -40,11 +42,35 @@ const stakeValidationMSG = defineMessage({
...
+    TokenActivationContext,
+  );
+
+  if (stakeType === 'tokens') {
+    return (
+      <Button
+        text={stakeValidationMSG.tokens}
+        textValues={{
+          leftToActivate: errorValues?.leftToActivate as PrimitiveType,
+          tokenSymbol: 'A',
```
Comment (ArmandoGraterol) [inline src/modules/dashboard/components/ActionsPage/StakingValidationError/StakingValidationError.tsx:56]: Just noticed, this should be dynamic right?

**Review by ArmandoGraterol — COMMENTED** [review]
With these styles the button looks good to me:
```
    display: flex;
    align-items: center;
    border: 1px solid transparent;
```
FireShot Capture 604 - Mint 1000 PRR - Motion - Colony - wonker - localhost

I added them to `.StakeRequiredBanner_share_ZIjAQMVz button`. The flex props aligns it and the `border` stops the button from moving around when the hover state is active.

File: src/modules/dashboard/components/ActionsPage/StakingValidationError/StakingValidationError.tsx (hunk 4 above)
Comment (chinins) [reply src/modules/dashboard/components/ActionsPage/StakingValidationError/StakingValidationError.tsx:56]: great catch! leftover from the working phase :) fixed now

**Comment by chinins** [comment]
> display: flex;
>     align-items: center;
>     border: 1px solid transparent;

Thank you so much! That was very helpful. I had to make border-color pink for the button not to move. Otherwise the style from the Button component intervene and it does move.

**Review by danbr — CHANGES_REQUESTED** [review]
Well done looks good. Tooltip is correctly located.
Screenshot 2022-06-07 at 11 46 07

But i think the space under the staking buttons should be reduced:
Screenshot 2022-06-07 at 11 55 52

Vs figma:

Screenshot 2022-06-07 at 12 02 54


Although its in figma, I feel that the share button is too close to the edge. But thats just personal preference.
Screenshot 2022-06-07 at 11 40 17

**Comment by chinins** [comment]
> But i think the space under the staking buttons should be reduced: Screenshot 2022-06-07 at 11 55 52
> 
> Vs figma:
> 
> Screenshot 2022-06-07 at 12 02 54


The figma is not consistent with the current dapp. This is how it is in master

Screenshot 2022-06-07 at 14 40 53

**Comment by chinins** [comment]
And as for the alignment - it's aligned with the hovered state of the avatar, 
Screenshot 2022-06-07 at 14 43 56

I can move the share url button 2 px left to have it aligned with avatar in the non-hover normal state. @arrenv , this is more a question to you.

**Review by arrenv — APPROVED** [review]
Look good and works well, nice one!

Regarding the alignment of the share button, it is aligned to the edge of the container, so, I feel it is more correct.

**Review by ArmandoGraterol — APPROVED** [review]
Great work! The button is correctly aligned for me and the rest is looking good too

**Comment by danbr** [comment]
> > But i think the space under the staking buttons should be reduced: 
> > Vs figma:
> 
> The figma is not consistent with the current dapp. This is how it is in master
> 
> Screenshot 2022-06-07 at 14 40 53
In that case, as there is no need for further updates, I am happy to approve. Well done.

**Review by danbr — APPROVED** [review]

**Pull request merged**